exercises/transpose/transpose_test.go

```diff
@@ -0,0 +1,41 @@
+package transpose
+
+import (
+	"fmt"
+	"math"
+	"testing"
+)
+
+// For a more visual representaion of the test data than is provided by
+// cases_test.go have a look at the original json data here:
+// https://github.com/exercism/x-common/blob/master/exercises/transpose/canonical-data.json
+
+func TestTranspose(t *testing.T) {
+	for _, test := range testCases {
+		actual := Transpose(test.input)
+		if fmt.Sprintf("%s", actual) != fmt.Sprintf("%s", test.expected) {
+			// find the shorter of actual and expected, to avoid index out of range errors
+			min := int(math.Min(float64(len(test.expected)), float64(len(actual))))
```

Comment:
No, I didn't see that SO question, I think I like the min function better though.